docs/design_docs/20230403-search_by_pk.md

```diff
@@ -0,0 +1,77 @@
+# MEP(Milvus Enhancement Proposals) Templates
+
+Current state: Under Discussion
+
+ISSUE: [[Feature]: Support to search by primary keys #23184](https://github.com/milvus-io/milvus/issues/23184)
+
+Keywords: Search, ANN
+
+Released: v2.3.0
+
+## Summary(required)
+
+Support to search (ANNS) by the query vectors corresponding to the given primary keys.
+
+## Motivation(required)
+
+For now, Milvus requires passing the query vectors to do anns, we have to fetch the vectors first if they are in the collection, which is complex and slow.
+
+We need a way to do anns directly for the corresponding vectors of given primary keys, which should be more efficient.
+
+## Public Interfaces(optional)
+
+Add new field `primary_keys` in `SearchRequest`, all SDKs adds new method:
```

Comment:
I think it is better to reuse the search interface and add an ids parameter. It is better to choose one of the two parameters id and vector

Reply:
Milvus still uses Search method, but the SDK need a new method (maybe expect languages like python, which could pass any number of parameters). Changing the SDKs' search method signature will break present code